firestore/client.go

```diff
@@ -381,3 +399,25 @@ func (ec emulatorCreds) GetRequestMetadata(ctx context.Context, uri ...string) (
 func (ec emulatorCreds) RequireTransportSecurity() bool {
 	return false
 }
+
+// ReadOptions provides specific instructions for how to access documents in the database.
+// Currently, only ReadTime is supported.
+type ReadOptions struct {
```

Comment:
Same comment as the datastore PR. Could we start by just adding a `WithReadTime`?

If we plan for many more read options, we could also do this:

``` golang
func (c *Client) WithReadOptions(ro ...ReadOption) *Client { /* ... */ }
```

which alleviates the burden on the caller to initialize a struct correctly.

Reply:
Done.